### ISSUE TITLE
Better support for custom Redpanda config

### ISSUE THREAD

**Issue by vuldin** [redpanda-data/redpanda-ansible-collection#69]
Is there another potential method for custom config aside from using a JSON dictionary?

https://docs.redpanda.com/current/deploy/deployment-option/self-hosted/manual/production/production-deployment-automation/#custom-configuration

I understand JSON/YAML are effectively the same thing, but it seems counterintuitive to use JSON when the rest of our deployment configs are yml along with the actual redpanda config files which are on the various servers. In my case I’ve embedded an escaped json in a central vars file I’m using for the install:

```
# json dict for custom configuration #
redpanda: "{\"cluster\":{\"kafka_enable_authorization\":\"true\",\"enable_sasl\":\"true\",\"superusers\":\"['admin,console,kfk-connect001']\",\"sasl_kerberos_keytab\":\"/var/lib/redpanda/kfk-rnd-rp.keytab\",\"sasl_mechanisms\":\"['SCRAM,GSSAPI']\",\"sasl_kerberos_principal\":\"redpanda\"}}"
```

**Comment by hcoyote** [comment]
JSON is used because the YAML merging is broken under Ansible. Gene looked at this a fair amount and determined this is a limitation in core Ansible. There's not much we can do about it short of ripping out the entire config management system and finding a different way to do it.

It's unlikely we're going to change this. 

cc @gene-redpanda

**Comment by gene-redpanda** [comment]
Well... you say that but I'd actually like to rip it out entirely and replace it with rpk, go templates and a lot of flags. I'm not quite sure how yet but I have it as something in the back of my head for next year. 

I'm also considering ripping out most of the advanced config and dumping it in yet another template and then using a light base template that handles node specific config and can merge with either our templates OR a user provided template. 

Bottom line I actually kind of agree with this one 😅 

That said you're absolutely right about the issue here. Python's yaml interpreter wasn't willing to support conditional statements in yaml maps which was a non starter as a lot of our optional elements needed to be exactly that!

**Comment by gene-redpanda** [comment]
Added support for per host overrides and various other fixes. Still need to use JSON but the issue is improved
